src/main/groovy/com/twcable/grabbit/jcr/JCRNodeDecorator.groovy

```diff
@@ -105,6 +106,37 @@ class JCRNodeDecorator {
         return definition.isMandatory()
     }
 
+    void checkoutIfNecessary(javax.jcr.Node node) throws RepositoryException {
```

Comment:
@jbornemann : I took a look at Sling's implementation for the same and this code is on same lines.. I still need to refactor it. I am holding off onto that because with protobuf update PR and https://github.com/TWCable/grabbit/pull/135, there would be changes in same classes and if I make changes now, I will have to redo those...